tripy/examples/resnet50/resnet50.ipynb

```diff
@@ -0,0 +1,383 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# ResNet50 Image Classificiation\n",
+    "\n",
+    "This notebook aims to demonstrate the implementation of a ResNet50 model using `tripy`. It implements the architecture, \n",
+    "loads pretrained weights, and runs predictions on a sample dataset.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "Install required libaries, load the image dataset for inference and pretrained resnet50 torch weights."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install datasets matplotlib pillow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "from transformers import ResNetForImageClassification, AutoImageProcessor\n",
+    "from datasets import load_dataset\n",
+    "\n",
+    "# Load the pretrained ResNet50 model from Hugging Face\n",
+    "resnet_pretrained = ResNetForImageClassification.from_pretrained(\"microsoft/resnet-50\")\n",
+    "resnet_pretrained.eval()\n",
+    "resnet_pretrained = resnet_pretrained.to('cuda')\n",
+    "\n",
+    "# Load the image processor\n",
+    "processor = AutoImageProcessor.from_pretrained(\"microsoft/resnet-50\")\n",
+    "\n",
+    "# Load a sample image from the dataset\n",
+    "dataset = load_dataset(\"comet-team/coco-500\", split=\"train\")\n",
+    "idx = 103\n",
+    "image = dataset[idx]['Image']\n",
+    "image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load the image processor\n",
+    "processor = AutoImageProcessor.from_pretrained(\"microsoft/resnet-50\")\n",
+    "\n",
+    "# Run the pre-trained PyTorch model on image as a sanity check for weights\n",
```

Comment:
I think we can probably drop the sanity check. It feels more like testing code than something we are trying to demonstrate.

Reply:
Fair enough, I removed it!